File: src/main/java/fr/marstech/mtlinkspray/entity/DashboardCategory.java
package fr.marstech.mtlinkspray.entity;

import lombok.*;
import lombok.experimental.Accessors;
import lombok.experimental.SuperBuilder;

import java.util.List;

@EqualsAndHashCode(callSuper = true)
@Accessors(chain = true)
@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class DashboardCategory extends DashboardItem {

    @Builder.Default
    private List<DashboardLink> links = List.of();
}

File: src/main/java/fr/marstech/mtlinkspray/entity/DashboardCategory.kt
package fr.marstech.mtlinkspray.entity

import java.util.*

data class DashboardCategory(
    override var id: String = UUID.randomUUID().toString(),
    override var name: String,
    override var description: String?,

    val links: MutableList<DashboardLink?> = mutableListOf<DashboardLink?>()
) : DashboardItem
